TextFormat.md

```diff
@@ -43,43 +38,379 @@ support more human-readable representations, but never at the cost of accurate r
+| `loop` | `loop` *label* `{` … `}` | `loop $a { br $a; }`
+| `if` | `if` `(` *condition* `)` `{` … `}` | `if (0) { 1 }`
+| `if` | `if` `(` *condition* `) `{` … `} else `{` … `}` | `if (0) { 1 } else { 2 }`
+| `br` | `br` *label* | `br $where`
+| `br` | `br` `(` *expr* `)` *label* | `br ($v) $where`
+| `br_if` | `br_if` `(` *expr* `)` *label* | `br_if ($x < $y) $where`
+| `br_if` | `br_if` `(` *expr* `,` *condition* `)` *label* | `br_if ($v, $x < $y) $where`
+| `br_table` | `br_table` `(` *index-expr* `)` `[` *label* `,` … `]` `,` *default-label* | `br_table ($i) [$somewhere, $or_other], $default`
+| `br_table` | `br_table` `(` *expr* `,` *index-expr* `)` `[` *label* `,` … `]` `,` *default-label* | `br_table ($v, $i) [$somewhere, $or_other], $default`
+| `return` | `return` | `return`
+| `return` | `return` *expr* | `return $x`
+| `unreachable` | `unreachable` | `unreachable`
+
+
+### Basic operators ([described here](https://github.com/WebAssembly/design/blob/master/AstSemantics.md#constants))
+
+| Name | Syntax | Example
+| ---- | ---- | ---- |
+| `i32.const` | … | `234`, `0xfff7`
+| `i64.const` | … | `234`, `0xfff7`
+| `f64.const` | … | `0.1p2`, `infinity`, `nan:0x789`
+| `f32.const` | … | `0.1p2`, `infinity`, `nan:0x789`
+| `get_local` | *name* | `$x + 1`
+| `set_local` | *name* `=` *expr* | `$x = 1`
+| `select` | `select` *expr*, *expr*, *expr* | `select 1, 2, $x < $y`
+| `call` | `call` *name* `(`*expr* `,` … `)` | `call $min(0, 2)`
+| `call_import` | `call_import` *name* `(`*expr* `,` … `)` | `call_import $max(0, 2)`
+| `call_indirect` | `call_indirect` *signature-name* `[` *expr* `] (`*expr* `,` … `)` | `call_indirect $foo [1] (0, 2)`
+
+### Arithmetic operators ([described here](AstSemantics#32-bit-integer-operators))
+
+Arithmetic operators use C/JS-like infix and prefix notation.
+
+Add, sub, mul, div, rem, and, or, xor, shl, and shr operators use
+`+`, `-`, `*`, `/`, `%`, `&`, `|`, `^`, `<<`, and `>>`, respectively.
+
+`eq`, `ne`, `lt`, `le`, `gt`, `ge`, operators use
+`==`, `!=`, `<`, `<=`, `>`, `>=`, respectively.
+
+`eqz` and `neg` operators use prefix unary `!` and `-`, respectively.
+
+Signed and unsigned operators are suffixed with `s` and `u`, respectively,
+so `i32.div_s` looks like `/s`, and so on.
+
+These operators use C/JS-style operator precedence rules, except that in order
+to correct [an old mistake](http://www.lysator.liu.se/c/dmr-on-or.html)),
+it is not valid to rely on precedence alone between a `&`, `|`, or `^` operator
+and another operator of similar precedence. Parentheses are required in such
+contexts in order to ensure that the code reads as intended.
+
+All other arithmetic operators use their actual name in a prefix notation,
+such as `f32.sqrt(…)`, with comma-seperated operands.
+
+## Module-level syntax
+
+Type declarations:
+
+`type $ternary of function (i32,i32,i32) : (i32)`
+
+Imports:
+
+`import "print" as $print_int from "global.PrintingStuff" typeof function (f64) : ()`
```

Comment:
Not a big fan of this syntax, but I suppose there's plenty of time to bikeshed over the specifics. :)